### PR TITLE
Paginate web users

### PR DESCRIPTION
**Pull request by biyeun**
screen shot 2015-01-05 at 6 50 24 pm

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
does this need any QA?

**Comment by czue** [comment]
(guessing @esoergel or @twymer might be best to review)

File: corehq/apps/users/static/users/js/web_users_list.ng.js [hunk 1]
```diff
@@ -9,36 +9,74 @@
...
+                }
+            }
+        };
+
+        self.retry = function () {
+            if (self.retries < 10) {
+                self.retries ++;
+                self.getUsers();
+            } else {
+                // show error
```
Comment (esoergel) [inline corehq/apps/users/static/users/js/web_users_list.ng.js]: `# TODO`

File: corehq/apps/users/templates/users/web_users.b3.html [hunk 2]
```diff
@@ -121,65 +131,145 @@
...
+            <div class="panel-heading">
+                <div class="row">
+                    <div class="col-sm-6">
+                        <h3 class="panel-title" style="padding-top: 7px;">{% trans 'Current Users' %}</h3>
+                    </div>
+                    <div class="col-sm-6">
+                        <div class="input-group">
+                            <input type="text"
+                                   class="form-control"
+                                   ng-model="query"
```
Comment (esoergel) [inline corehq/apps/users/templates/users/web_users.b3.html]: that's cool.  This'll just automagic the input field to the `query` var?

File: corehq/apps/users/static/users/js/web_users_list.ng.js [hunk 3]
```diff
@@ -9,36 +9,74 @@
...
 
+        $scope.query = '';
+
+        $scope.error = null;
+        $scope.notLoaded = true;
+
+        self.retries = 0;
+
+        self.updateUserList = function (data) {
+            if (data.success) {
```
Comment (esoergel) [inline corehq/apps/users/static/users/js/web_users_list.ng.js]: Is this success condition different from the one used below? (line 69)

File: corehq/apps/users/templates/users/web_users.b3.html [hunk 4]
```diff
@@ -121,65 +131,145 @@
...
+                            <th>{% trans "Phone Numbers" %}</th>
+                            <th>{% trans "Remove" %}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr ng-show="notLoaded">
+                            <td colspan="5">{% trans "Loading Users..." %}</td>
+                        </tr>
+                        <tr ng-repeat="user in paginatedItems">
+                            {% angularjs %}
```
Comment (esoergel) [inline corehq/apps/users/templates/users/web_users.b3.html]: How's this work again?  all the ``{{ vars }}`` inside this tag aren't rendered by the django template, so angular takes over?

File: corehq/apps/users/templates/users/web_users.b3.html [hunk 5]
```diff
@@ -250,7 +340,7 @@ <h3 class="panel-title">
                         <h4 class="modal-title">{% trans "Delete invitation" %}</h4>
                     </div>
                     <div class="modal-body">
-                        <h4> {% trans "Are you sure you want to delete this invitation?" %}</h4>
+                        <p>{% trans "Are you sure you want to delete this invitation?" %}</p>
                     </div>
                     <div class="modal-footer">
                         <button type="button" class="btn btn-default" data-dismiss="modal">{% trans "Cancel" %}</button>
```
Comment (esoergel) [inline corehq/apps/users/templates/users/web_users.b3.html]: I think this is my first time reading angular stuff - it's remarkably straightforward.

File: corehq/apps/users/views/__init__.py [hunk 6]
```diff
@@ -370,10 +377,30 @@ class NewListWebUsersView(BaseUserSettingsView):
...
+                "default_operator": "AND",
+                "fields": default_fields if is_simple else None
+            }},
+            "filter": {"and": web_user_filter},
+            "sort": {'username.exact': 'asc'},
+        }
+        return es_query(
+            params={}, q=q, es_url=ES_URLS["users"],
+            size=limit, start_at=skip,
+        )
```
Comment (esoergel) [inline corehq/apps/users/views/__init__.py]: Probably not worth changing, but FWIW you can use
```python
q = (UserES()
     .domain(self.domain)
     .web_users()
     .sort("username.exact")
     .set_query({
         "query_string": {
             "query": query,
             "default_operator": "AND",
             "fields": default_fields if is_simple else None}})
     .start(skip)
     .size(limit))

return q.run()
```

File: corehq/apps/users/views/__init__.py [hunk 7]
```diff
@@ -382,15 +409,66 @@ def web_users(self):
...
 
     @property
     @memoized
     def user_roles(self):
         user_roles = [AdminUserRole(domain=self.domain)]
-        user_roles.extend(sorted(UserRole.by_domain(self.domain),
-                                 key=lambda role: role.name if role.name else u'\uFFFF'))
+        user_roles.extend(sorted(
+            UserRole.by_domain(self.domain),
+            key=lambda role: role.name if role.name else u'\uFFFF'
```
Comment (esoergel) [inline corehq/apps/users/views/__init__.py]: what does the escape sequence do here?  Is it just `<` everything else or something?

**Comment by esoergel** [comment]
I've got a couple questions, but nothing blocking.  Looks good!

**Pull request merged**